### PR TITLE
Allow enable_shielded_nodes to be false

### PR DESCRIPTION
**Pull request by ScottSuarez**
closes https://github.com/hashicorp/terraform-provider-google/issues/9118

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
container: fixed bug where `enable_shielded_nodes` could not be false on resource `google_container_cluster`
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -1407,7 +1407,11 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 		ResourceLabels: expandStringMap(d, "resource_labels"),
 	}
 
-	if v, ok := d.GetOk("enable_shielded_nodes"); ok {
+	// shielded nodes is computed and optional yet serverside
+	// default is true. Forcing true here esentially serves
+	// as a default false but is unavoidable due to how
+	// computed and GetOk work together.
+	if v, _ := d.GetOk("enable_shielded_nodes"); true {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_container_cluster.go.erb:1414]: Can we just remove the check altogether and use `d.Get`?

**Review by rileykarson — COMMENTED** [review]
It's probably worth amending the docs and release note to state that Terraform will now default to `false` on creation regardless of the GKE version in use.

@c2thorn: Could force-sending `false` here cause any issues with autopilot?

**Comment by c2thorn** [comment]
> @c2thorn: Could force-sending `false` here cause any issues with autopilot?

There's [containerClusterAutopilotCustomizeDiff](https://github.com/hashicorp/terraform-provider-google-beta/blob/master/google-beta/resource_container_cluster.go#L3981) which will automatically set `enable_shielded_nodes` as true for autopilot. Removing the `ok` check shouldn't affect it

File: mmv1/third_party/terraform/resources/resource_container_cluster.go.erb (hunk 1 above)
Comment (ScottSuarez) [reply mmv1/third_party/terraform/resources/resource_container_cluster.go.erb]: Yep, I suppose I wanted to leave the if statement in there to identify that it was irregular and forcibly divergent behavior, but the comment serves that purpose just fine

**Comment by ScottSuarez** [comment]
@rileykarson the documentation already states the default is false
https://github.com/hashicorp/terraform-provider-google/blob/master/website/docs/r/container_cluster.html.markdown#L160

**Review by rileykarson — APPROVED** [review]
Ah great, thanks!

**Pull request merged**